experiments/mp-game-6/draw.creatures.js

```diff
@@ -6,29 +6,48 @@
 //       multiplayer game 6.
 // --------------------------------------------------------------------
 
-function drawBoxTable(numButtons, numLights, train, roundProps) {
-    console.log("drawBoxTable has been reached")
-    var lights = ""
+function drawBoxTable(numButtons, numLights, numStars, train, roundProps) {
+    var lights = []
     for(var i = 1; i<=numLights; i++){
-        lights = lights + "<div id='light" + i + "' style='width:125px;padding:50px 0px;background-color:teal; margin:10px;" +
-                      " opacity:0.5;text-align:center;font-size:25px;border-radius:50%;display:inline-block'>light " + i + "</div>";
-        console.log("<div id='light" + i + "' style='width:125px;padding:50px 0px;background-color:teal; margin:50px;" +
-                    " opacity:0.5;text-align:center;font-size:25;border-radius:50%'>light " + i + "</div>")
+        let ran_Num = Math.floor(Math.random()*lights.length)
+        var light = "<div id='light" + i + "' style='width:125px;padding:50px 0px;background-color:" + color(i) +"; margin:10px;" +
+                " opacity:0.5;text-align:center;font-size:25px;border-radius:50%;display:inline-block'>light " + i + "</div>";
+        lights.splice(ran_Num, 0, light)
```

Comment:
I would first just add the lights to the array in any order with `lights.push(light)`, then randomly shuffle the array.

This is better because (1) in general, checking properties of an array as you loop through it is ripe for errors, and (2) `splice` is generally an unclear function, and is ripe for bugs (as you've found out!)

For a shuffle function (which may be useful in other places as well), just paste in this code from [this StackOverflow answer](https://stackoverflow.com/questions/6274339/how-can-i-shuffle-an-array)

```javascript
/**
 * Shuffles array in place.
 * @param {Array} a items An array containing the items.
 */
function shuffle(a) {
    var j, x, i;
    for (i = a.length - 1; i > 0; i--) {
        j = Math.floor(Math.random() * (i + 1));
        x = a[i];
        a[i] = a[j];
        a[j] = x;
    }
    return a;
}
```